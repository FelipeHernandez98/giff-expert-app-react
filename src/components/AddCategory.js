import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ( {setCategories} ) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        //Viene el evento onChage y se modifica con el hook 
        setInputValue(e.target.value);

        console.log('Handle input change llamado')
    }
   

    const handleSubmit = (e)=>{
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue, ...cats ])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue }
                onChange={handleInputChange}//Es el evento que determina un cambio y captura el cambio
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
