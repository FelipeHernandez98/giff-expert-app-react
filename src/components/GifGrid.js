import React from 'react'

export const GifGrid = ({category}) => {

    const getGifs = async() =>{
        
        const url = 'https://api.giphy.com/v1/gifs/search?q=Dragon Ball&limit=10&api_key=7lN5ubz0G3FO13uMmuww5jJKYtoW0B8b';
        const resp = await fetch(url);
        const {data} = await resp.json();

        const gifs = data.map(img =>{
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
    }

    getGifs();

    return (
        <>
            <h3>{ category }</h3>
        </>
    )
}
