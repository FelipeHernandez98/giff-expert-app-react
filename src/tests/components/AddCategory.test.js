import React from 'react';
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('pruebas en el <AddCategory />', () => {

    const setCategories = ()=>{};
    const wrapper = shallow(<AddCategory setCategories={ setCategories }/>);

    test('el componente se renderiza correctamente', () => {

        expect( wrapper).toMatchSnapshot();

    });

    test('debe cambiar la caja de texto', () => {

        const input = wrapper.find('input');
        const value = 'Hola mundo';

        input.simulate('change', { target: { value } });

        expect( wrapper.find('p').text().trim()).toBe( value )
    })
})