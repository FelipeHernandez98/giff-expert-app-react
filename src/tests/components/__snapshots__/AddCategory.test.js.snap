// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <AddCategory /> el componente se renderiza correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
